src/ui/public/index_patterns/index_patterns.js

```diff
@@ -41,6 +41,7 @@ export function IndexPatternsProvider(Notifier, Private) {
   self.cache = patternCache;
   self.getIds = getProvider('id');
   self.getTitles = getProvider('attributes.title');
+  self.getIndexPatterns = getProvider();
```

Comment:
Personally, I think that this is an example of abstraction getting in the way. Is there any reason to keep it? I think we should rip it out and replace it with this:

```js
import { cloneDeep } from 'lodash'
function IndexPatternsProvider() {
  // ...

  const wrapWithPromiseCache = (get) => {
    let promise;
    return () => {
      promise = promise || get()
      return promise.then(cloneDeep) // always from the cached promise clone
    }
  }

  self.getIds = wrapWithPromiseCache(() => {
    return self.find([], 10000).then(resp => (
      resp.savedObjects.map(obj => obj.id)
    ))
  });

  self.getTitles = () => {
    return self.find(['title'], 10000).then(resp => (
      resp.savedObjects.map(obj => obj.attributes.title)
    ))
  };

  self.getSavedObjects = (fields, count) => {
    return savedObjectsClient.find({
      type: 'index-pattern',
      fields,
      perPage: count
    })
  };
}
```